runtime/src/bank.rs

```diff
@@ -2500,12 +2497,28 @@ impl Bank {
     }
 
     pub fn deposit(&self, pubkey: &Pubkey, lamports: u64) {
-        let mut account = self.get_account(pubkey).unwrap_or_default();
-        self.collected_rent.fetch_add(
-            self.rent_collector.update(pubkey, &mut account),
-            Ordering::Relaxed,
-        );
```

Comment:
Well, there were two bugs..:

1. When a new account is created via `.deposit()` at epoch N, its rent is calculated since epoch 0. (this was the original bug this pr fixed)
1. Also, the exact timing of the rent collection is wrong: we should add `lamports` then collect rents inside this `fn`. `self.rent_collector.update(Account::default())` (= new account case) doesn't collect rents. And it effectively stores the account without collecting rent, it only adds lamports. So actual rent collection happens the next time the account is touched.. (I've found this bug today..)

Reply:
This is fixed by not doing rent collection at all as the new behavior.